### DISCUSSION TITLE
Keeping track of form values, even after onSubmit

### DISCUSSION THREAD

**Discussion by brandondurham** [react-hook-form/react-hook-form#6024] [Q&A]
I’m trying to keep track of all fields and their values, but it seems like `onSubmit` resets that state or something, even though the form is still filled and valid. I’ve tried using `getValues()` and `watch`, but with both of those I first get something like this:
```js
{email: "tipsy_tassles@smallparade.com", password: "123456", firstName: "T", lastName: "T"}
```
… then as soon as I hit “Submit”, I get:
```js
{email: undefined, password: undefined, firstName: undefined, lastName: undefined}
```
Is it possible to get access to the form fields and their values even after hitting “Submit”?

**Comment by bluebill1049** [comment]
I don't think we reset the value. can you share a codesandbox?

**Reply by brandondurham** [reply]
Oooooohhhhhh. Damn. It’s because I’m disabling all fields while `isSubmitting === true`.

https://codesandbox.io/s/relaxed-kapitsa-zuw0k

I updated the forms to move `disabled={ isSubmitting }` to the fieldset and it works now. My preference is that it applies to the individual fields as I appreciate the visual feedback that the fields are disabled during submit, but having it actually work is clearly more important.
